t/job-manager/sched-helper.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/sh
+#
+
+# job-manager sched helper functions
+
+get_state() {
+	local id=$1
+	local state=$(${LIST_JOBS} | awk '$1 == "'${id}'" { print $2; }')
+	test -z "$state" \
```

Comment:
Suggestion given that these are separated from the test script:
* define LIST_JOBS in `sched-helper.sh` rather than depending on it being defined externally.
* consider more descriptive names and flagging functions that are not for external use like `get_state`
* add comment for each function explaining what it does and how to call it
* document that the annotation functions require HAVE_JQ in callers

Reply:
I was actually going to ask the `for try` loops in here are really necessary?
It would be good to understand why they are needed (perhaps it is obvious, sorry), since it indicates a potential use case for synchronizing with jobs in which there is a known race condition.

In this case `get_state` is synchronizing with the `clean` event and fetching job state directly from the job-manager (AFAICS), so the need for the try loop isn't clear.

In `check_annotation` it is perhaps a little more obvious why a try loop is required, since an annotation update can come at any time, so there is no "event" on which to synchronize. However, for future editors of the testsuite, a comment may not hurt. ;-)

Reply:
> define LIST_JOBS in sched-helper.sh rather than depending on it being defined externally.

Thanks missed that!

Reply:
> In this case get_state is synchronizing with the clean event and fetching job state directly from the job-manager (AFAICS), so the need for the try loop isn't clear.

TBH I didn't know either, as it was the code that was there beforehand.  I suspect the loop is there just in case the state transition happens a little slowly (although there should be a sleep for 0.5 in there, I'll add it).

Going through logs, it appears the `clean` event check is in  95bddec68de67145fe20c9bda70d9013fb1ba894.  If the job goes inactive and is no longer listed by the job-manager, the logic is to report the job state as inactive.  I'll add a comment to that effect.

Reply:
Yeah, I was thinking we might want to revisit these tests, but I wasn't sure I understood all the details (even though they are my creation from the early days of the job manager - sorry).  For now some comments would be a big improvement over what I left you with.

Reply:
Ok, I didn't realize these were not new functions. Sorry about that!

Reply:
There _are_ new functions, but they mimic the old ones.  I just didn't want to ask @chu11 to clean up my mess, though he's welcome to :-)